### ISSUE TITLE
Error attempting to make Stripe payment using IPFS through Brave

### ISSUE THREAD

**Issue by gullible1** [OriginProtocol/dshop#8]
**Submitted by Protocol Labs:**

When attempting to complete a purchase through the Brave Swag store at roughly 2200EST 21APR2020 while accessing it through a local IPFS node using the IPFS Companion, it errored when attempting to complete the payment. I was attempting to make a ~$6 purchase using a Mastercard CC shipping to a valid US address.

Brave Desktop (Mac) v 1.5.123
go-ipfs v 0.4.23
IPFS Companion v 2.11.0

**Initial Guess:**

Could be that Stripe.js doesn't like that the live connection doesn't appear to be over HTTPS. The Security tab reports: "This page has a non-HTTPS secure origin"

**URI:** 

http://localhost:8080/ipns/store.brave.com/#/order/QmNX3pv55134tQnatD6KD7ueKfdpKNVDbDCf52p9wqTEbL?auth=2c2dd03bljc61fc3rg911p05w6449dj

**Console Log:**

DomDistiller debug level: 0
0.da54ef82.bundle.js:121 Invalid asm.js: Unexpected token
(index):1 You may test your Stripe.js integration over HTTP. However, live Stripe.js integrations must use HTTPS.
value @ (index):1
5.da54ef82.bundle.js:1 Make Stripe payment request...
(index):1 You may test your Stripe.js integration over HTTP. However, live Stripe.js integrations must use HTTPS.
value @ (index):1
inpage.js:1 Heads up! ethereum._metamask exposes methods that have not been standardized yet. This means that these methods may not be implemented in other dapp browsers and may be removed from MetaMask in the future.
get @ inpage.js:1
5.da54ef82.bundle.js:1 Make Stripe payment request...

**Snapshot:**
![Screen Shot 2020-04-21 at 10 24 45 PM](https://user-images.githubusercontent.com/46481405/81420352-73a21700-9182-11ea-8684-18579fb4a54e.png)

**Comment by mikeshultz** [comment]
I thought @nick was already digging into this one?

**Comment by DanielVF** [comment]
The submitter is correct! 

Stripe.js won't work in production over HTTP.

We really just need to add a check if we are using a live stripe key, and are not in HTTPs. If so, we need to disable the credit card option, possibly with an explanation.

**Comment by mikeshultz** [comment]
I think there was other options for this that @nick was looking into.  Hosted payment pages/forwards/etc.  There's bound to be a way to pull this off rather than just disabling it, even if it means just hosting a payment page on the backend for this edge case or configuring a different CC processor.

**Comment by nick** [comment]
Yea I'm thinking we open a window to take CC payment over HTTPS direct from the server, then post the result back to the client with postMessage

**Comment by mikeshultz** [comment]
Stripe does appear to have hosted checkout pages:

https://stripe.com/docs/payments/checkout
https://github.com/stripe-samples/checkout-one-time-payments

https://github.com/stripe-samples/checkout-one-time-payments/blob/4a60e58c7549ef499aea9b009cabcf1ed02982b1/client-only/client/react-cra/src/components/Checkout.js#L75-L80

Probably worth exploring

**Comment by nick** [comment]
Yep could do it that way too... though I guess we'd have less control of the checkout experience in that case. Might be quicker to implement though 🤷‍♂️

**Comment by DanielVF** [comment]
I just recently setup stripe checkout for a project and it was very easy to use.

**Comment by mikeshultz** [comment]
I very much like not handling CC data if I can avoid it.  Push that liability to someone with better compliance and insurance.  

The edge case folks wanting to use IPFS in a distributed fashion would probably be okay with their configured Stripe checkout.  Might be worth asking the Protocol Labs folks since they brought it up?  I don't see this as much more than a fallback.

**Comment by nick** [comment]
When I mentioned "open a window to take CC payment over HTTPS direct from the server, then post the result back to the client with postMessage" I meant by using Stripe.js... so CC details would not be sent to our server (which is a whole can of worms anyway and not really practical).

The idea is:
- Checkout UI detects if user is on https or not. For non-https, hide the credit card entry form and make button say 'Proceed to payment' 
- When user clicks that button, we open a popup pointing to, eg, https://checkout.ogn.app and POST the cart details and shop id
- https://checkout.ogn.app shows a UI for entering credit card details which the user fills out
- Once payment has been made, popup window uses postMessage to tell the parent window the payment was successful, then closes itself
- User is forwarded to confirmation screen

We could also do the above with redirects instead of in a popup.
Or... we could just use the Stripe checkout. Or do both. 🤷‍♂️

**Comment by mikeshultz** [comment]
ah, yeah, I misunderstood, that should work just fine.  I don't have a strong preference for either.

**Comment by franckc** [comment]
Doing dashboard cleanup. https://github.com/orgs/OriginProtocol/projects/3
OK if I move this to the Backlog given we are not actively working on it right now?

**Comment by gullible1** [comment]
Sure, especially since we won't be working on the Protocol Labs store for a while.

**Comment by franckc** [comment]
Downgrading to P2 so that only external deployer launch related bugs are labelled as P0 / P1.
